pinot-perf/src/main/java/org/apache/pinot/perf/BenchmarkFuseRegexp.java

```diff
@@ -0,0 +1,343 @@
+
+  @Benchmark
+  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  public void decreasing9Regex(Blackhole bh) {
+    String query = "SELECT INT_COL FROM MyTable WHERE "
+        + "regexp_like(DOMAIN_NAMES, '^.domain9.*$') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain8.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain7.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain6.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain5.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain4.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain3.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain2.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain2.*')";
+    bh.consume(getBrokerResponse(query));
+  }
+
+  @Benchmark
+  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  public void decreasing9Fusing(Blackhole bh) {
+    String query = "SELECT INT_COL FROM MyTable WHERE "
+        + "regexp_like(DOMAIN_NAMES, '"
+        + "(?:^.*domain9.*$)|"
+        + "(?:^.*domain8.*$)|"
+        + "(?:^.*domain7.*$)|"
+        + "(?:^.*domain6.*$)|"
+        + "(?:^.*domain5.*$)|"
+        + "(?:^.*domain4.*$)|"
+        + "(?:^.*domain3.*$)|"
+        + "(?:^.*domain2.*$)|"
+        + "(?:^.*domain1.*$)')";
+
+    bh.consume(getBrokerResponse(query));
+  }
+
+  @Override
+  protected String getFilter() {
+    return null;
+  }
+
+  @Override
+  protected IndexSegment getIndexSegment() {
+    return _indexSegment;
+  }
+
+  @Override
+  protected List<IndexSegment> getIndexSegments() {
+    return Collections.singletonList(_indexSegment);
+  }
+
+  public static void main(String[] args)
+      throws Exception {
+    new Runner(new OptionsBuilder().include(BenchmarkFuseRegexp.class.getSimpleName())
+//        .addProfiler(JavaFlightRecorderProfiler.class)
+        .build()).run();
+  }
+
+  private void checkCorrectness() {
+    String query1 = "SELECT INT_COL FROM MyTable WHERE "
+        + "regexp_like(DOMAIN_NAMES, '^.domain9.*$') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain8.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain7.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain6.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain5.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain4.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain3.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain2.*') or "
+        + "regexp_like(DOMAIN_NAMES, '^.domain2.*')";
+    String query2 = "SELECT INT_COL FROM MyTable WHERE "
+        + "regexp_like(DOMAIN_NAMES, '"
+        + "(?:^.*domain9.*$)|"
+        + "(?:^.*domain8.*$)|"
+        + "(?:^.*domain7.*$)|"
+        + "(?:^.*domain6.*$)|"
+        + "(?:^.*domain5.*$)|"
+        + "(?:^.*domain4.*$)|"
+        + "(?:^.*domain3.*$)|"
+        + "(?:^.*domain2.*$)|"
+        + "(?:^.*domain1.*$)')";
+
+    long total1 = getBrokerResponse(query1).getTotalDocs();
+    long total2 = getBrokerResponse(query2).getTotalDocs();
+
+    if (total1 != total2) {
```

Comment:
This is not the correct check. This returns the total documents of the table, instead of the matching docs. `getNumDocsScanned` will return the documents matched